azurelinuxagent/common/protocol/wire.py

```diff
@@ -840,25 +839,36 @@ def get_ext_conf(self):
                 self.ext_conf = ExtensionsConfig(xml_text)
         return self.ext_conf
 
-    def get_ext_manifest(self, ext_handler, goal_state):
+    def get_ext_manifest(self, ext_handler, incarnation):
+
+        def get_local_file(name, incarnation):
+            local_file = MANIFEST_FILE_NAME.format(name, incarnation)
+            return os.path.join(conf.get_lib_dir(), local_file)
+
+        try:
+            local_file = get_local_file(ext_handler.name, incarnation)
```

Comment:
I think this adds complexity. If you feel the right thing to do is check the disk cache first, add that into the code below.

Reply:
If the ExtHandlersHandler is processing Goal State N, I think the right thing to do is to check the disk cache for Goal State N. I think that checking the disk cache for Goal State N+1 is the wrong thing to do. If the disk cache has an entry for a manifest labeled with N+1 (and implicitly, not for Goal State N), it is more likely than not that something weird happened.

That said, taking another look, I think I do see a way to simplify this a bit.